### DISCUSSION TITLE
Renovate doesn't create PRs if checks are not available

### DISCUSSION THREAD

**Discussion by JoergHeinicke5005** [renovatebot/renovate#19306] [Request Help]
We are running Renovate self-hosted with following config:
```
module.exports = {
  "extends": [
    "config:base",
    ":combinePatchMinorReleases",
    ":separateMultipleMajorReleases",
    ":assignAndReview(team:kr)",
    ":enableVulnerabilityAlerts",
    ":label(renovate)",
    ":prNotPending",
    ":renovatePrefix",
    ":updateNotScheduled",
    ":automergeBranch",
    ":automergeMinor",
    ":automergeRequireAllStatusChecks",
  ],
  "ignorePresets": ["group:monorepos", "group:recommended"],
  "docker": {"pinDigests": true },
  "commitMessagePrefix": "[renovate]",
  "dependencyDashboardAutoclose": true,
  "exposeAllEnv": true,
  "onboarding": false,
  "platform": "github",
  "requireConfig": "optional",
  "schedule": null,
  "hostRules": [
    ...
  ],
  "packageRules": [
    ...
  ],
};
```
For some changes in our repos we don't have automated checks set up, Renovate creates a pull request and doesn't automerge as expected. We can manually go through the PRs and apply them quickly.

But then again we have several cases where this is not happening, but instead Renovate stops after only creating the branch and updating the dashboard. No PR is being create, not easy workflow for us to follow. And I can't figure out why, the log files don't give much insights:
```
DEBUG: getBranchPr(renovate/java-java-dependencies) (repository=oec/kvm)
DEBUG: findPr(renovate/java-java-dependencies, undefined, open) (repository=oec/kvm)
DEBUG: findPr(renovate/java-java-dependencies, undefined, closed) (repository=oec/kvm)
DEBUG: branch.isBehindBase(): using cached result "false" (repository=oec/kvm)
DEBUG: isBranchConflicted(main, renovate/java-java-dependencies) (repository=oec/kvm)
DEBUG: branch.isConflicted(): using cached result "false" (repository=oec/kvm)

DEBUG: getBranchPr(renovate/gradle-gradle-7.x) (repository=oec/kvm)
DEBUG: findPr(renovate/gradle-gradle-7.x, undefined, open) (repository=oec/kvm)
DEBUG: findPr(renovate/gradle-gradle-7.x, undefined, closed) (repository=oec/kvm)
DEBUG: branch.isBehindBase(): using cached result "false" (repository=oec/kvm)
DEBUG: isBranchConflicted(main, renovate/gradle-gradle-7.x) (repository=oec/kvm)
DEBUG: branch.isConflicted(): using cached result "false" (repository=oec/kvm)
```
Any hints on what keeps Renovate from creating the PRs? Any additional information required?

**Comment by rarkins** [comment]
It's because of your `:prNotPending` setting

**Reply by JoergHeinicke5005** [reply]
Thanks for the response. 2 questions though:
1) Exactly the same gradle upgrade lead to a PR being created on another repository:
![image](https://user-images.githubusercontent.com/53432510/206389323-c636897d-0840-459c-9dc8-3ead19057086.png)
This is why I'm wondering what's the difference between the 2 repos.
2) On the dashboard in the section "Pending Status Checks" it lists the Gradle upgrade. In addition, the text reads "To force their creation now, click the checkbox below." From that I understand, checking the checkbox would create a PR nonetheless - but this has no effect for the first repo either. Is the text not quite exact here? Should the checkbox overrule :prNotPending?

**Reply by rarkins** [reply]
1. You would need to either look at the status checks in the repo, or check Renovate's debug logs. Ultimately, if there's no status checks and you have prCreation=not-pending, you shouldn't get a PR created
2. This might be a bug if it's not working

**Reply by JoergHeinicke5005** [reply]
As mentioned I do have cases where a PR for the gradle upgrade was created despite no status check on the particular upgrade and prCreation=not-pending.
As you can see, for the renovate/gradle-gradle-7.x branch no checks have been performed:
![image](https://user-images.githubusercontent.com/53432510/206863719-958be189-0ff9-4b5a-87d7-ce5ba5d84a7d.png)
On the PR it's also visible that there are no checks but it did get created by Renovate Bot (bottom of screenshot):
![image](https://user-images.githubusercontent.com/53432510/206863903-743ad3b6-4697-4cd6-b027-d65af51a36c6.png)
Since I can't figure out what's the difference between our repositories, I was wondering what makes it work in the one and not work in the other. Unfortunately, the debug logs of a current run on that repository are no longer helpful since the PR had already been created. The Renovate run with which the PR was created was not in debug mode.

Also, if prCreation=not-pending did prevent the PRs from being created, would there be a setting with which I can get PRs created even without checks? From the documentation I do understand prCreation=immediate would do that but I prefer not to have the noise of always creating PRs. IMHO, both the aspects don't seem to be linked with each other: Why not getting a PR created with prCreation=not-pending when there are no checks (in contrary to prCreation=status-success where this is understandable from the documentation)?

**Reply by viceice** [reply]
renovate will open the PR after 24h after branch creation  because of missing checks

**Reply by JoergHeinicke5005** [reply]
> renovate will open the PR after 24h after branch creation because of missing checks

Awesome, thanks for the response! I can confirm it for a test repo I created.
Can this be documented on [prCreation](https://docs.renovatebot.com/configuration-options/#prcreation)?

Unfortunately, I can't mark it as answer as you just responded on this thread.

**Reply by JoergHeinicke5005** [reply]
I gave it a shot: https://github.com/renovatebot/renovate/pull/19345
